### PR TITLE
adding equations for MR and GP

### PR DESCRIPTION
**Pull request by karenlstengel**

### CODE REVIEW

**Comment by jeremylt** [comment]
Note - I recommend rebasing this branch off of main, as the CI infrastructure has been significantly overhauled.

**Comment by jedbrown** [comment]
I rebased the branch.

There were some minor formatting issues that I've fixed, and expanded the Mooney-Rivlin section with more explanation (but still work-in-progress). To preview while I'm working, I use the virtualenv instructions at the bottom of `doc/README.md`. Actually, after the first time, I use `make doc DOXYGEN=true` to avoid re-running doxygen, thereby making the incremental rebuild much faster.

File: examples/solids/index.rst [hunk 1]
```diff
@@ -340,14 +340,64 @@ Carrying through the differentiation :math:numref:`strain-energy-grad` for the m
...
+      \Phi(\mathbb{\bar I_1}, \mathbb{\bar I_2}, J) = \frac{\mu_1}{2} (\mathbb{\bar I_1} - 3) + \frac{\mu_2}{2} (\mathbb{\bar I_2} - 3) + \frac{k_1}{2} (J - 1)^2,
+
+   which we differentiate as in the Neo-Hookean case :math:numref:`neo-hookean-stress` to yield the second Piola-Kirchoff tensor,
+
+   .. math::
+      :label: mooney-rivlin-stress
+
+      \bm S = \mu_1 \mathbb I_1 J^{-2/3}(\bm I_3 - \frac{1}{3} \bm C^{-1}) + \mu _2 I_2J^{-4/3}(I_1 - \bm C - \frac{2}{3} \bm C^{-1}) + k_1(J^2 -J)\bm C^{-1} ,
+
+   [TODO: double-check above] where we have used
```
Comment (jedbrown) [inline examples/solids/index.rst:374]: I didn't mess with this equation, but if the new text above and below is correct, it'll need to be adjusted. (I may well have bungled something so please double-check me.)

File: examples/solids/index.rst [hunk 2]
```diff
@@ -340,14 +340,64 @@ Carrying through the differentiation :math:numref:`strain-energy-grad` for the m
    For example, if :math:`e_{ij} \sim 10^{-8}`, then naive computation of :math:`\bm I_3 - \bm C^{-1}` and :math:`\log J` will have a relative accuracy of order :math:`10^{-8}` in double precision and no correct digits in single precision.
    When using the stable choices above, these quantities retain full :math:`\varepsilon_{\text{machine}}` relative accuracy.
 
+.. admonition:: Mooney-Rivlin model
+   :class: dropdown
```
Comment (jedbrown) [inline examples/solids/index.rst:344]: These blocks render as
image
so we can include an adequate derivation without being disruptive to the narrative.

**Comment by rezgarshakeri** [comment]
It is not finished yet, we need to update the Jacobian

**Pull request closed without merging**